src/TSTransformer/macros/propertyCallMacros.ts

```diff
@@ -90,55 +90,49 @@ const STRING_CALLBACKS: MacroList<PropertyCallMacro> = {
 		}),
 };
 
-const ARRAY_LIKE_METHODS: MacroList<PropertyCallMacro> = {
-	size,
-};
-
-const READONLY_ARRAY_METHODS: MacroList<PropertyCallMacro> = {
-	isEmpty: (state, node, expression) =>
-		lua.create(lua.SyntaxKind.BinaryExpression, {
-			left: lua.create(lua.SyntaxKind.UnaryExpression, { operator: "#", expression }),
-			operator: "==",
-			right: lua.number(0),
-		}),
-
-	join: (state, node, expression) => {
-		const separator = node.arguments.length > 0 ? transformExpression(state, node.arguments[0]) : lua.string(", ");
-		return lua.create(lua.SyntaxKind.CallExpression, {
-			expression: lua.globals.table.concat,
-			args: lua.list.make(expression, separator),
-		});
-	},
-
-	every: (state, node, expression) => {
+function makeEveryOrSomeMethod(
+	iterator: lua.Identifier,
+	callbackArgsListMaker: (
+		keyId: lua.TemporaryIdentifier,
+		valueId: lua.TemporaryIdentifier,
+		expression: lua.Expression,
+	) => lua.List<lua.Expression>,
+	hasInvertedCallback: boolean,
+	startBool: boolean,
+	endBool: boolean,
+): PropertyCallMacro {
+	return (state, node, expression) => {
 		expression = state.pushToVarIfComplex(expression);
 
-		const resultId = state.pushToVar(lua.bool(true));
+		const resultId = state.pushToVar(lua.bool(startBool));
 		const callbackId = state.pushToVarIfComplex(transformExpression(state, node.arguments[0]));
 
-		const keyId = lua.tempId();
-		const valueId = lua.tempId();
+		const firstId = lua.tempId();
+		const secondId = lua.tempId();
```

Comment:
I think keyId, valueId were more descriptive names for these. I can't think of a scenario where we're using something other than pairs/ipairs.

Reply:
Set uses it the opposite way around. ValueId, ContainsId. Initially wrote it for set and then converted it into this function, never changed the names back.